src/pocketmine/entity/Monster.php

```diff
@@ -20,8 +20,52 @@
 */
 
 namespace pocketmine\entity;
-
+use pocketmine\event\entity\EntityDamageByEntityEvent;
+use pocketmine\Player;
+use pocketmine\network\protocol\AddEntityPacket;
+use pocketmine\item\Item as ItemItem;
 
 abstract class Monster extends Creature{
-
+	
+	// Add a createAddEnityPacket method so don't have to replicate this code in all of the subclasses
+	
+	protected function createAddEntityPacket(Player $player){
+		$pk = new AddEntityPacket();
+		$pk->eid = $this->getId();
+		$pk->x = $this->x;
+		$pk->y = $this->y;
+		$pk->z = $this->z;
+		$pk->speedX = $this->motionX;
+		$pk->speedY = $this->motionY;
+		$pk->speedZ = $this->motionZ;
+		$pk->yaw = $this->yaw;
+		$pk->pitch = $this->pitch;
+		$pk->metadata = $this->dataProperties;
+		return $pk;
```

Comment:
I just copied this code from Zombie.php function spawnTo - I noticed it was the same code over and over again, so thought it would be better to create a method to hold the packet creation and place in a higher level class.  I am not sure how to use clone here.